### PR TITLE
EVG-6389: make SSH keys configurable

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-6389

This allows us to add SSH keys other than the key maintained by Chef. New SSH key pairs can be added in the admin config (they can't be edited/removed though). Keys are updated on an hourly basis (except for the local app server's keys, which are only updated when the app server restarts).

I'm keeping the SSH key field in the distro settings for backwards compatibility with the existing SSH key. Once we add other SSH keys, we can get rid of this.

## Admin Settings
* Add SSH key pairs and the directory containing them (on the app servers) to the admin settings. The EC2 regions that have the SSH key are awkwardly in the SSH key struct, but there wasn't much of a better place to put them.
* Drive-by fix for `*evergreen.Settings`, which would be saved without `ValidateAndDefault()`, so default values would not be set.
* Drive-by fix for AWS region settings.

## Updating Local SSH Keys
* Write the private key into every single app server.
* When attempting to SSH, use all of the available private key files in the SSH key directory (including the default mci one we currently use).

## Updating Static Host Keys
* Add a line to the authorized keys file on running static hosts (only if the line isn't present).

## Updating Cloud Provider SSH Keys
* Add cloud manager functionality to import an SSH key.

### CODE REVIEW

File: model/host/hostutil.go [hunk 1]
```diff
@@ -137,27 +138,26 @@ func (h *Host) GetSSHInfo() (*util.StaticHostInfo, error) {
...
-// single distro-level SSH key name corresponding to an existing SSH key file on
-// the app servers. We should be able to handle multiple keys configured in
-// admin settings rather than from a file name in distro settings.
 func (h *Host) GetSSHOptions(settings *evergreen.Settings) ([]string, error) {
-	keyPath, err := h.GetSSHKeyPath(settings)
-	if err != nil {
-		return nil, errors.WithStack(err)
+	var keyPaths []string
+	for _, pair := range settings.SSHKeyPairs {
+		if _, err := os.Stat(pair.PrivatePath(settings)); err == nil {
```
Comment (bsamek) [inline model/host/hostutil.go]: Should we handle the error case?

File: model/host/host.go [hunk 2]
```diff
@@ -2188,3 +2214,18 @@ func FindHostWithVolume(volumeID string) (*Host, error) {
...
+// as those in the global settings.
+func FindNeedsNewSSHKeys(settings *evergreen.Settings) ([]Host, error) {
+	names := []string{}
+	for _, pair := range settings.SSHKeyPairs {
+		names = append(names, pair.Name)
+	}
+
+	return Find(db.Query(bson.M{
+		StatusKey:      evergreen.HostRunning,
+		ProviderKey:    evergreen.ProviderNameStatic,
```
[the remark this answers is not in the document]
Comment (bsamek) [reply model/host/host.go]: Ah, found it later. Maybe change the function name for clarity.

File: units/ssh_keys_update_local.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/pkg/errors"
+)
+
+const localUpdateSSHKeysJobName = "update-ssh-keys-local"
+
+type localUpdateSSHKeysJob struct {
```
Comment (bsamek) [inline units/ssh_keys_update_local.go]: This job is quite risky. We should come up with a deployment strategy. We could lock ourselves out of every app server and build host.

Comment (Kimchelly) [reply units/ssh_keys_update_local.go]: The greatest risk is probably if you happen to overwrite our current SSH key that's the key maintained by Chef. I added more sanity checking to `(*evergreen.Settings) ValidateAndDefault()` so we can be sure that updating the SSH keys won't delete or modify existing keys (including the Chef SSH key).

This job runs every time an app server restarts so if for some reason the local SSH key gets deleted, we can just restart and it'll be back after this job runs.

**Pull request merged**